src/PublicKey.ts

```diff
@@ -44,4 +44,8 @@ export class PublicKey {
             pub: new Buffer(this.key.data),
         });
     }
+
+    public getType(): KeyType {
+        return this.key.type;
+    }
```

Comment:
This is also an API change; I'd run this past Todd, but this is a change that 1) is an addition and 2) is not implementation specific.  So I think I'm fine with it

Reply:
I've updated the initial PR description to note the API changes.